File: audiobook/src/main/java/de/ph1b/audiobook/persistence/BookmarkTable.java
package de.ph1b.audiobook.persistence;

import android.content.ContentValues;
import android.database.sqlite.SQLiteDatabase;

import de.ph1b.audiobook.model.Bookmark;

/**
 * Collection of strings representing the bookmark table
 *
 * @author Paul Woitaschek
 */
class BookmarkTable {
    static final String PATH = "bookmarkPath";
    static final String TITLE = "bookmarkTitle";
    static final String TABLE_NAME = "tableBookmarks";
    static final String TIME = "bookmarkTime";
    private static final String BOOK_ID = "bookId";
    private static final String CREATE_TABLE = "CREATE TABLE " + TABLE_NAME + " ( " +
            PATH + " TEXT NOT NULL, " +
            TITLE + " TEXT NOT NULL, " +
            TIME + " INTEGER NOT NULL, " +
            BOOK_ID + " INTEGER NOT NULL, " +
            "FOREIGN KEY (" + BOOK_ID + ") REFERENCES " + BookTable.TABLE_NAME + "(" + BookTable.ID + "))";

    static ContentValues getContentValues(Bookmark bookmark, long bookId) {
        ContentValues cv = new ContentValues();
        cv.put(TIME, bookmark.getTime());
        cv.put(PATH, bookmark.getMediaFile().getAbsolutePath());
        cv.put(TITLE, bookmark.getTitle());
        cv.put(BookTable.ID, bookId);
        return cv;
    }


    public static void onCreate(SQLiteDatabase db) {
        db.execSQL(CREATE_TABLE);
    }

    public static void dropTableIfExists(SQLiteDatabase db) {
        db.execSQL("DROP TABLE IF EXISTS " + TABLE_NAME);
    }
}

File: audiobook/src/main/kotlin/de/ph1b/audiobook/persistence/BookmarkTable.kt
package de.ph1b.audiobook.persistence

import android.content.ContentValues
import android.database.sqlite.SQLiteDatabase
import de.ph1b.audiobook.model.Bookmark

/**
 * Represents an sql table for bookmarks
 *
 * @author Paul Woitaschek
 */
object BookmarkTable {

    const val PATH = "bookmarkPath"
    const val TITLE = "bookmarkTitle"
    const val TABLE_NAME = "tableBookmarks"
    const val TIME = "bookmarkTime"
    const val BOOK_ID = "bookId"
    private const val CREATE_TABLE = "CREATE TABLE $TABLE_NAME ( " +
            "  $PATH TEXT NOT NULL, " +
            "  $TITLE TEXT NOT NULL, " +
            "  $TIME INTEGER NOT NULL, " +
            "  $BOOK_ID INTEGER NOT NULL, " +
            "  FOREIGN KEY ( $BOOK_ID ) REFERENCES ${BookTable.TABLE_NAME} ( ${BookTable.ID} ) " +
            ")"

    fun getContentValues(bookmark: Bookmark, bookId: Long): ContentValues {
        val cv = ContentValues()
        cv.put(TIME, bookmark.time)
        cv.put(PATH, bookmark.mediaFile.absolutePath)
        cv.put(TITLE, bookmark.title)
        cv.put(BookTable.ID, bookId)
        return cv
    }

    fun onCreate(db: SQLiteDatabase) {
        db.execSQL(CREATE_TABLE)
    }

    fun dropTableIfExists(db: SQLiteDatabase) {
        db.execSQL("DROP TABLE IF EXISTS $TABLE_NAME")
    }
}